ably/rest_http_error.go

```diff
@@ -6,15 +6,16 @@ import (
 )
 
 type RestHttpError struct {
-	Msg      string
-	Response *http.Response
-
+	Msg          string
 	ResponseBody string
+
+	resp *http.Response
```

Comment:
As I already said, I'm going to rewrite RestHttpError in a separate PR, as it's useless in current form. Is that ok?